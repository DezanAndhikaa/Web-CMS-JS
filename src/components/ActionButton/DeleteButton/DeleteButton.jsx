import React from 'react';
import { Button } from '@material-ui/core';
import './DeleteButton.scss';

class DeleteButton extends React.PureComponent {
	render() {
		return (
			<div className="del-button-row">
				<div className="del-button" onClick={this.props.onClick} />
			</div>
		);
	}
}

export default DeleteButton;