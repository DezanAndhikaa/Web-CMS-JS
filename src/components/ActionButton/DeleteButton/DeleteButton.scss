@import 'themes/functions';
@import 'themes/variables';

.del-button-row{
    display: flex;
    justify-content: flex;
    .del-button{
        height: 35px;
        width: 35px;
        margin-left: vw(25);
        border-radius: vw(8) !important;
        box-shadow: 1px 2px 4px rgba(0, 0, 0, 0.1);
        background-image: url('assets/icons/ic_delete.png');
        background-color: white;
        background-repeat: no-repeat;
        background-size: 50%;
        background-position-x: 46%;
        background-position-y: center;
    }
}

@media (min-width: 600px) and (max-width: 1025px) {
    .del-button-row{
        display: flex;
        justify-content: flex-end;
        height: vw-tab(30);
        width: 100%;
        .del-button{
            height: vw-tab(30);
            width: vw-tab(30);
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            background-image: url('assets/icons/ic_delete.png');
            background-repeat: no-repeat;
            background-size: contain;
        }
    }
}

@media (max-width: 599px) {    
    .del-button-row{
        display: flex;
        justify-content: flex-end;
        height: vw-mobile(20);
        width: 100%;
        .del-button{
            height: vw-mobile(20);
            width: vw-mobile(20);
            box-shadow: 0px vw(3) vw(6) rgba(0, 0, 0, 0.161);
            background-image: url('assets/icons/ic_delete.png');
            background-repeat: no-repeat;
            background-size: contain;
        }
    }
}