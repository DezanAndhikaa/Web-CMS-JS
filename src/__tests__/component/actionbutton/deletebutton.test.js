import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import DeleteButton from "../../../components/ActionButton/DeleteButton/DeleteButton";

Enzyme.configure({adapter:new Adapter()});
it("render deletebutton correctly", () => {
    const wrapper = shallow (
      <DeleteButton />
    )
    expect(wrapper).toMatchSnapshot();

})