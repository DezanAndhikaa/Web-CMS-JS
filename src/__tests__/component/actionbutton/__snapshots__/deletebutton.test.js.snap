exports[`render delete button correctly 1`] = `ShallowWrapper {}`;